src/index.js

```diff
@@ -137,7 +137,9 @@ export default function promiseMiddleware(config = {}) {
        *   }
        * }
        */
-      return promise.then(handleFulfill, handleReject);
```

Comment:
>  [...] the promise is resolved before the actions are dispatched (and state updated), which is undesirable.

I would tend to disagree with this. The middleware is designed to dispatch actions that describe async changes in state. More specifically, the pending action describes a promise with a state of "unsettled," the fulfilled action describes a promise with a state of "resolved," and the rejected action describes a promise with a state of "rejected." I think that is clear. The reason I am bringing this up is, because the actions describe the state of the promise; promises are state machines, after all. Therefore, I would argue the fulfilled action should not be dispatched _until after_ the promise is in a resolved state. This is not "undesirable," but by design.